parlai/agents/transformer/polyencoder.py

```diff
@@ -227,14 +246,16 @@ def get_encoder(self, opt, dict, null_idx, reduction_type):
             output_scaling=opt['output_scaling'],
         )
 
-    def attend(self, attention_layer, queries, keys, mask):
+    def attend(self, attention_layer, queries, keys, values, mask):
         """ Unify the API of MultiHeadAttention and
-            BasicAttention that are slighlty different
+            BasicAttention that are slightly different
         """
-        if isinstance(attention_layer, BasicAttention):
-            return attention_layer(queries, keys, mask)
+        if keys is None:
+            keys = values
+        if isinstance(attention_layer, PolyBasicAttention):
```

Comment:
i see that it was like this before, but i find this a bit strange. can we set something like `self.attention_type` so we don't have to check the module name every time?

Reply:
I think the reason it is like this is that the codes version has two levels of attention, which can be different (i.e. there is a `self.codes_attention_type` as well), so we'd have to add another flag in here somewhere anyway